### ISSUE TITLE
out().<field> converts embedded json to array in OrientDB 2.1.10

### ISSUE THREAD

**Issue by anrudo** [orientechnologies/orientdb#5752]
This query:

```sql
select from V where label = 'snapshot' and ((out('derivedFrom').data.id contains 'e3e3a92a-8dd2-4963-89fd-8386cc2b0cf8') and (data.stage = 'dev'))
```
Yields a result in OrientDB 2.1.5, but fails for me in 2.1.10.

The reason is, that given a vertex with a field that contains an embedded json object, if one traverses the graph with the `out()` function, the object ends up being converted to an array, dropping the property names. 

To reproduce, create a two-vertex graph with an edge between them, give one vertex the field `label = "foo"` and the other one an embedded json field `data = {"bar": "baz", "quux": 1}`.

```sql
select out().data from (select from V where label = 'foo')
```

Yields `[{"bar": "baz", "quux": 1}]` in 2.1.5 but `["baz", 1]` in 2.1.10.

Is this behaviour intended?

**Comment by anrudo** [comment]
Hey, just wanted to mention that this is still unsolved in 2.1.16, cheers!

**Comment by luigidellaquila** [comment]
Hi @anrudo 

I just pushed a fix for this on both 2.1.x and develop. The fix will be available in 2.1.17 and 2.2 GA

I'm closing this issue, please reopen if you see any other problems

Thanks

Luigi

**Comment by anrudo** [comment]
@luigidellaquila thanks, very much appreciated!
